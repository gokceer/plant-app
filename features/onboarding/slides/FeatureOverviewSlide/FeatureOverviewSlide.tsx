import React from "react";
import { Dimensions, Image, StyleSheet, Text } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

const { width: screenWidth, height: screenHeight } = Dimensions.get("window");

const FeatureOverviewSlide = () => {
  return (
    <SafeAreaView>
      <Text style={styles.title}>
        Take a photo to
        <Text style={styles.titleBold}> identify </Text>
        {"\n"}the plant!
      </Text>
      <Image
        source={require("../../../../assets/images/featureOverview.png")}
        accessibilityLabel="Example of taking a photo to identify a plant"
        style={styles.image}
      />
    </SafeAreaView>
  );
};

export default FeatureOverviewSlide;

const styles = StyleSheet.create({
  title: {
    fontSize: 28,
    marginLeft: 24,
    marginVertical: 12,
    fontFamily: "RubikMedium",
  },
  titleBold: {
    fontFamily: "RubikBold",
  },
  image: {
    width: screenWidth,
    height: screenHeight * 0.75,
    resizeMode: "contain",
    alignSelf: "center",
  },
});
