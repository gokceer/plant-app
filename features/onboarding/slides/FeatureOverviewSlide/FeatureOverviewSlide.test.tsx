import { render } from "@testing-library/react-native";
import React from "react";
import FeatureOverviewSlide from "./FeatureOverviewSlide";

describe("FeatureOverviewSlide", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<FeatureOverviewSlide />);
    expect(toJSON()).toMatchSnapshot();
  });
});
