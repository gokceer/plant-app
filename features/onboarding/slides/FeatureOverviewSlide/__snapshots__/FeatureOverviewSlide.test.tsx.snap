// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureOverviewSlide renders correctly and matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "RubikMedium",
        "fontSize": 28,
        "marginLeft": 24,
        "marginVertical": 12,
      }
    }
  >
    Take a photo to
    <Text
      style={
        {
          "fontFamily": "RubikBold",
        }
      }
    >
       identify 
    </Text>
    

    the plant!
  </Text>
  <Image
    accessibilityLabel="Example of taking a photo to identify a plant"
    source={
      {
        "testUri": "../../../assets/images/featureOverview.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "height": 1000.5,
        "resizeMode": "contain",
        "width": 750,
      }
    }
  />
</RNCSafeAreaView>
`;
